extensions/sd_api_pictures/script.py

```diff
@@ -139,41 +156,74 @@ def bot_prefix_modifier(string):
 
     return string
 
-def force_pic():
-    global picture_response
-    picture_response = True
+def toggle_generation(*args):
+    global picture_response, shared, streaming_state
+    if not args:
+        picture_response = not picture_response
+    else:
+        picture_response = args[0]
+
+    shared.args.no_stream = True if picture_response else streaming_state # Disable streaming cause otherwise the SD-generated picture would return as a dud
+    shared.processing_message = "*Is sending a picture...*" if picture_response else "*Is typing...*"
+    btn_text = "Suppress the picture response" if picture_response else "Force the picture response"
+    
+    return btn_text
+
+def filter_address(address):
+    address = address.strip()
+    address = re.sub('http(s)?:\/\/|\/$','',address) # remove starting http:// OR https:// OR trailing slash
+    return address
+
+def SD_api_address_update(address):
+    
+    global params
+    
+    msg = "✔️ SD API is found on:"
+    address = filter_address(address)
+    params.update({"address": address})
+    try:
+        response = requests.get(url=f'http://{params["address"]}/sdapi/v1/sd-models')
+        response.raise_for_status()
+        r = response.json()
+    except:
+        msg = "❌ No SD API endpoint on:"
+
+    return gr.Textbox.update(label=msg)
+
+
 
 def ui():
 
     # Gradio elements
-    with gr.Accordion("Stable Diffusion api integration", open=True):
+    gr.Markdown('### Stable Diffusion API Pictures')
```

Comment:
The title is written bellow another h1 title "sd_api_pictures", which I guess is added automatically.